components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/dao/DBBasedTemplateManager.java

```diff
@@ -0,0 +1,351 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.email.mgt.dao;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.email.mgt.exceptions.I18nEmailMgtException;
+import org.wso2.carbon.email.mgt.internal.I18nMgtDataHolder;
+import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerServerException;
+import org.wso2.carbon.identity.governance.model.NotificationTemplate;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is responsible for managing the email templates in the database.
+ */
+public class DBBasedTemplateManager implements TemplatePersistenceManager {
+
+    private static final Log log = LogFactory.getLog(DBBasedTemplateManager.class);
+
+    private final NotificationTypeDAO notificationTypeDAO = new NotificationTypeDAO();
+    private final OrgNotificationTemplateDAO orgNotificationTemplateDAO = new OrgNotificationTemplateDAO();
+    private final AppNotificationTemplateDAO appNotificationTemplateDAO = new AppNotificationTemplateDAO();
+
+    @Override
+    public void addNotificationTemplateType(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        String templateTypeKey = displayName.toLowerCase();
+        int tenantId = getTenantId(tenantDomain);
+
+        notificationTypeDAO.addNotificationTemplateType(templateTypeKey, displayName, notificationChannel, tenantId);
+
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("%s template type: %s for tenant: %s successfully added.", notificationChannel,
+                    displayName, tenantDomain));
+        }
+    }
+
+    @Override
+    public boolean isNotificationTemplateTypeExists(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        String templateTypeKey = displayName.toLowerCase();
+        int tenantId = getTenantId(tenantDomain);
+
+        String templateTypeDisplayName =
+                notificationTypeDAO.getNotificationTemplateType(templateTypeKey, notificationChannel, tenantId);
+        boolean isEmailTemplateTypeExists = StringUtils.isNotBlank(templateTypeDisplayName);
```

Comment:
IMO renaming this to `isNotificationTemplateTypeExists` is better to avoid ambiguity and generalize